fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexScrubber.java

```diff
@@ -221,8 +228,21 @@ public Builder setEntriesScanLimit(long entriesScanLimit) {
                 return this;
             }
 
+            /**
+             * Declare that the scrubbed index is a value index, even if it's type's name is different.
+             *
+             * Scrubbing dangling index entries only makes sense for value indexes, where the index entry points back to
+             * the record.
+             * Typically, this function is called to allow scrubbing of an index of a user-defined type. If called, it
+             * is the caller's responsibility to verify that the scrubbed index matches the "value" type criteria.
```

Comment:
It's probably worth clarifying what the "value" criteria are so that callers can know whether it's safe for them to set this flag

Reply:
Changed to: 
If called, it is the caller's responsibility to verify that the scrubbed index matches the one-to-one index-records criteria.
Would that do?

Reply:
Hm, I think it maybe could use a little more wordsmithing. I think the actual requirement is:

1. For the dangling scrubber job, every index entry needs to contain the primary of the record that generated it so that we can detect if that record is present
2. For the missing entry scrubber, the index key for the record needs to be present in the index

And then there's maybe room to clarify what exactly is meant by the "index key" (e.g., does it get generated by evaluating the index expression against the record or is there some other way for the index maintainer to generate the index keys) or how the primary key gets extracted from each entry. I think this might be what you mean by "one-to-one", but that's also a little vague

Reply:
1. (did you mean "primary key of the record"?) 
2. I'm afraid that do not understand what you mean.. (shouldn't the index maintainer of any type, if produces any index entry to update, always provide its key?)

Reply:
1. Ah, yes, that should be primary _key_ of the record
2. Not necessarily. Part of the problem here is that there are a few different "keys" that are relevant to this, but I'm going to use "FDB key" to mean "byte array that is the key of a key-value pair in FDB" and "index key" to mean "the result of calling `.getKey()` on an `IndexEntry` object". In a value index, those two things are very closely related: for each index entry returned by `filteredIndexEntries`, there is one FDB key per index entry, and it is derived by combining the index key with the record's primary key in [`FDBRecordStoreBase::indexEntryKey`](https://github.com/foundationdb/fdb-record-layer/blob/2eadf38f86b5f58f806852c40bea821135feeba5/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java#L1046). That's the behavior that the scrubber assumes as part of its validation, but index maintainers are free to do whatever they want, and some have wildly different relationships between FDB and index keys. For example, the atomic mutation indexes have one FDB key written per _group_ (i.e., the first few columns of the index key), and text and lucena indexes explode text data (extracted from a single index key) to many different FDB keys.

Reply:
✅ updated